
The alsa-driver source is intended to be 2.6 only - all 2.4 compat cruft
is placed in .patch files and applied on the fly. However in the latest
version not all apply - removing the non-applying hunks make the package
compile ...

  - Rene Rebe <rene@exactcode.de>

diff -ur alsa-driver-1.0.7/pcmcia/pdaudiocf/pdaudiocf.patch alsa-driver-1.0.7-fixed/pcmcia/pdaudiocf/pdaudiocf.patch
--- alsa-driver-1.0.7/pcmcia/pdaudiocf/pdaudiocf.patch	2004-07-27 14:35:05.000000000 +0200
+++ alsa-driver-1.0.7-fixed/pcmcia/pdaudiocf/pdaudiocf.patch	2004-11-22 19:55:48.880901416 +0100
@@ -19,18 +19,6 @@
  #include "pdaudiocf.h"
  #include <sound/initval.h>
  
-@@ -57,7 +62,11 @@
- MODULE_PARM_SYNTAX(enable, SNDRV_ENABLE_DESC);
- module_param(irq_mask, int, 0444);
- MODULE_PARM_DESC(irq_mask, "IRQ bitmask for " CARD_NAME " soundcard.");
-+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 0)
-+MODULE_PARM(irq_list, "1-4i");
-+#else
- module_param_array(irq_list, int, boot_devs, 0444);
-+#endif
- MODULE_PARM_DESC(irq_list, "List of Available interrupts for " CARD_NAME " soundcard.");
-  
- 
 @@ -75,6 +84,17 @@
  static int pdacf_event(event_t event, int priority, event_callback_args_t *args);
  static void snd_pdacf_detach(dev_link_t *link);
diff -ur alsa-driver-1.0.7/pcmcia/vx/vxpocket.patch alsa-driver-1.0.7-fixed/pcmcia/vx/vxpocket.patch
--- alsa-driver-1.0.7/pcmcia/vx/vxpocket.patch	2004-04-07 19:53:41.000000000 +0200
+++ alsa-driver-1.0.7-fixed/pcmcia/vx/vxpocket.patch	2004-11-22 19:56:38.590344432 +0100
@@ -8,18 +8,6 @@
  #include "vxpocket.h"
  #include <sound/initval.h>
  
-@@ -72,7 +73,11 @@
- MODULE_PARM_SYNTAX(enable, SNDRV_ENABLE_DESC);
- module_param(irq_mask, int, 0444);
- MODULE_PARM_DESC(irq_mask, "IRQ bitmask for " CARD_NAME " soundcard.");
-+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 0)
-+MODULE_PARM(irq_list, "1-4i");
-+#else
- module_param_array(irq_list, int, boot_devs, 0444);
-+#endif
- MODULE_PARM_DESC(irq_list, "List of Available interrupts for " CARD_NAME " soundcard.");
- module_param_array(ibl, int, boot_devs, 0444);
- MODULE_PARM_DESC(ibl, "Capture IBL size for " CARD_NAME " soundcard.");
 @@ -176,3 +181,5 @@
  
  module_init(init_vxpocket);
